pages/entryListPage.js

```diff
@@ -11,40 +14,87 @@ export default function EntryListHome() {
     }
   );
 
+  const [editingEntry, setEditingEntry] = useState(null);
+
   const addJournalEntry = (newJournalEntry) => {
     setJournalEntries((oldJournalEntries) => [
       { id: nanoid(), ...newJournalEntry },
       ...oldJournalEntries,
     ]);
   };
+
+  function handleEditJournalEntry(editedEntry) {
+    setJournalEntries((oldJournalEntries) =>
+      oldJournalEntries.map((journalEntry) => {
+        if (journalEntry.id === editingEntry.id) {
+          return {
+            ...journalEntry,
+            ...editedEntry,
+          };
+        }
+        return journalEntry;
+      })
+    );
+    setEditingEntry(null);
+  }
+
   function handleDeleteJournalEntry(id) {
-    setJournalEntries(
-      journalEntries.filter((journalEntry) => journalEntry.id !== id)
+    setJournalEntries((oldJournalEntries) =>
+      oldJournalEntries.filter((journalEntry) => journalEntry.id !== id)
     );
   }
+
+  function handleOpenEditForm(id) {
+    const entryToEdit = journalEntries.find((entry) => entry.id === id);
+    setEditingEntry(entryToEdit);
+  }
+
+  function handleCloseEditForm() {
+    setEditingEntry(null);
+  }
+
   return (
     <>
-      <body>
-        <StyledHeader>List of Entries</StyledHeader>
-        <ul>
-          {journalEntries.map((journalEntry) => (
+      <StyledHeader>List of Entries</StyledHeader>
+      <section>
+        {journalEntries &&
+          journalEntries.map((journalEntry) => (
             <StyledEntry key={journalEntry.id}>
-              <h4>{journalEntry.destination}</h4>
+              <h4>Destination: {journalEntry.destination}</h4>
               <div>
-                from {journalEntry.from} to {journalEntry.to}
+                From {journalEntry.from} to {journalEntry.to}
               </div>
-              <h5>Titel: {journalEntry.entryTitle}</h5>
+              <h5>Title: {journalEntry.entryTitle}</h5>
               <p>{journalEntry.entry}</p>
-              <button
+              <StyledButton
                 type="button"
                 onClick={() => handleDeleteJournalEntry(journalEntry.id)}
               >
-                delete
-              </button>
+                Delete
+              </StyledButton>
+              <StyledButton
+                type="button"
+                onClick={() => handleOpenEditForm(journalEntry.id)}
+              >
+                Edit
+              </StyledButton>
             </StyledEntry>
           ))}
-        </ul>
-      </body>
+      </section>
+
+      {editingEntry && (
+        <EditFormWrapper>
+          <StyledHeader>Edit Entry</StyledHeader>
+          <FormComponent
+            onSubmitEvent={handleEditJournalEntry}
+            onCancelEvent={handleCloseEditForm}
```

Comment:
in your form component you don't expect the two props initialEntry and onCancelEvent. I guess you can leave these lines out.

Reply:
Otherwise you need to get the initialEntry into your form. Right now the initial state of your city is coming from the router. You might switch that to your initialEntry prop.

Reply:
Hey Felix, 
I have adjusted everything according to your comments. I just I left the initialEntry, hope thats fine.
 Thank you for your review!!